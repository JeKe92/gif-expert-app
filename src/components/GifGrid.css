.gifs-grid {
    display: grid;
    gap: 20px;
    grid-template-columns: repeat(4, 1fr);

}
@media screen and ( max-width: 960px ) {
    .gifs-grid {
        grid-template-columns: repeat(3, 1fr);
    }
}
@media screen and ( max-width: 768px ) {
    .gifs-grid {
        grid-template-columns: repeat(2, 1fr);
    }
}
@media screen and ( max-width: 550px ) {
    .gifs-grid {
        grid-template-columns: repeat(1, 1fr);
    }
}