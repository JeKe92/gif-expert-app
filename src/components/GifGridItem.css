.gifs-card{
    align-self: center;
    border: 2px solid #eeeeee;
    border-radius: 5px;
    height: auto;
    justify-self: center;
    min-height: 100%;
    width: 100%;    
}
.gifs-card>img {
    display: block;
    height: 200px;
    margin-left:auto;
    margin-right: auto;
    max-width: 100%;
}
.gifs-card>figcaption {
    text-align:center;
    padding: 10px;
    text-transform: capitalize;
}
