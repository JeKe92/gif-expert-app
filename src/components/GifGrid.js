import React from 'react';
import PropTypes from 'prop-types';
import { useFetchHook } from '../hooks/useFetchHook';
import './GifGrid.css';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchHook(category);    

    return (
        <article>
            <h3 className='animate__animated animate__fadeInLeft'>{category} {loading && <small>"Cargando..."</small>}</h3>
            <div className="gifs-grid">
            {images.map( gif => <GifGridItem key={gif.id} {...gif}/>)}
            </div>
        </article>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};