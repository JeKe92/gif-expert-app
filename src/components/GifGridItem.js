import React from 'react';
import PropTypes from 'prop-types';

import './GifGridItem.css';

export const GifGridItem = ( { title, url } ) => (
            <figure className="gifs-card animate__animated animate__fadeInDown">
                            <img src={url} alt={title}></img>
                            <figcaption>{title}</figcaption>
            </figure>
);

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}