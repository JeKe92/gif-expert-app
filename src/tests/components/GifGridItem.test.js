import React from 'react';
import { shallow } from "enzyme";

import {GifGridItem} from "../../components/GifGridItem";

describe('Tests for GridGifItem', () => {
    const title = 'My title';
    const url = 'https://localhost:3000/img/my-img.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    test('should show <GifGridItem />', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should show the title in the p tag', () => {
        const figcaption = wrapper.find('figcaption');
        expect( figcaption.text() ).toBe( title );
    });

    test('should have the alt as the title and src as the url for img tag', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe( url );
        expect(img.prop('alt')).toBe( title );
    });
    
    test('figure should have the card class', () => {
        const figure = wrapper.find('figure');
        expect(figure.prop('className')).toContain('gifs-card');
    });
    
    
    
});
