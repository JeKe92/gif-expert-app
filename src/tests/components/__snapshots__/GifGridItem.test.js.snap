// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GridGifItem should show <GifGridItem /> 1`] = `
<figure
  className="gifs-card animate__animated animate__fadeInDown"
>
  <img
    alt="My title"
    src="https://localhost:3000/img/my-img.jpg"
  />
  <figcaption>
    My title
  </figcaption>
</figure>
`;
